plumbing/clock/testing.go

```diff
@@ -0,0 +1,52 @@
+package clock
+
+import (
+	"time"
+
+	"github.com/benbjohnson/clock"
+)
+
+// MockBlockClock defines a testing structure used to mock a BlockClock.
+type MockBlockClock struct {
+	blockClock *clock.Mock
+	blockTime  time.Duration
+}
+
+// NewMockBlockClock returns a new MockBlockClock with the provided blockTime.
+// Calls to MockBlockClock.EpochSeconds() will return 0 until a time is set using
+// the method SetClock().
+func NewMockBlockClock(blkTime time.Duration) *MockBlockClock {
```

Comment:
I suggest taking the mock time value in this constructor, saving most users a call to `SetClock`